lib/dao.js

```diff
@@ -2365,6 +2575,184 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replace = function replace(data, options, cb) {
+  var self = this;
+  var connectionPromise = stillConnecting(this.getDataSource(), this, arguments);
+  if (connectionPromise) {
+    return connectionPromise;
+  }
+
+  if (options === undefined && cb === undefined) {
+    if (typeof data === 'function') {
+      cb = data;
+      data = undefined;
+    }
```

Comment:
As mentioned above, I don't think this API usage (`replace(callback)`) makes sense, no need to support it.